batch/batch/worker/worker.py

```diff
@@ -816,6 +819,8 @@ async def _cleanup(self):
             if self.netns:
                 network_allocator.free(self.netns)
                 self.netns = None
+
+            await blocking_to_async(self.pool, shutil.rmtree, self.container_scratch, ignore_errors=True)
```

Comment:
Isn't the Job the one that creates the scratch directory? I would think that whoever creates the directory should be the one to destroy it.